### PR TITLE
[NSX] Add ACL types support

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

This PR:
- Creates a group for the NSX segment on its creation
- Adds a Distributed Firewall policy for a segment along with its rules
- Removes the Distributed Firewall policy for the segment before its deleted

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

File: plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/utils/NsxControllerUtils.java [hunk 1]
```diff
@@ -45,6 +45,10 @@ public static String getNsxNatRuleId(long domainId, long accountId, long dataCen
         return String.format("D%s-A%s-Z%s-%s%s-NAT", domainId, accountId, dataCenterId, resourcePrefix, resourceId);
     }
 
+    public static String getNsxDistributedFirewallPolicyRuleId(String segmentName, long ruleId) {
+        return String.format("%s-P%s", segmentName, ruleId);
```
Comment (Pearl1594) [inline plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/utils/NsxControllerUtils.java:49]: I think we decided to use R to identify rules

**Review by Pearl1594 — APPROVED** [review]

File: plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/utils/NsxControllerUtils.java (hunk 1 above)
Comment (nvazquez) [reply plugins/network-elements/nsx/src/main/java/org/apache/cloudstack/utils/NsxControllerUtils.java]: Fixed, thanks

**Comment by nvazquez** [comment]
@blueorangutan package

**Pull request merged**